### PR TITLE
test: Add tests for key rotation and root publishing

### PR DESCRIPTION
**Pull request by asraa**
#### Summary

See title

#### Ticket Link

Part of https://github.com/sigstore/root-signing/issues/237

#### Release Note

```release-note

```

### CODE REVIEW

File: tests/e2e_test.go [hunk 1]
```diff
@@ -287,3 +291,253 @@ func TestSnapshotUnvalidatedFails(t *testing.T) {
...
+	sort.Strings(expectedKeyIds)
+	sort.Strings(actualKeyIds)
+	if !cmp.Equal(expectedKeyIds, actualKeyIds) {
+		t.Fatalf("expected key IDs %s, got %s", expectedKeyIds, actualKeyIds)
+	}
+
+	// Expect version 2 for root.
+	if root.Version != 2 {
+		t.Fatalf("expected root version 2, got %d", root.Version)
+	}
```
Comment (Hayden-IO) [inline tests/e2e_test.go:547]: Should we also test verification for rotations? There's two things we can test:
* With keys 1 and 2, rotate out 2 for 3, sign with 1, and test that you can still verify
* With keys 1 and 2, rotate out 1 for 3, and sign with 2, and test you can verify

Comment (asraa) [reply tests/e2e_test.go:547]: I think I can parametrize what key to rotate out for and add verification, can add!

Comment (Hayden-IO) [reply tests/e2e_test.go:547]: Feel free to do this in another PR too if it'll change non-test code.

**Comment by trishankatdatadog** [comment]
Glad to see this PR. I believe @hosseinsia and I wrote and reviewed the go-tuf root rotation code extensively, and would certainly love to see tests to make sure the code meets the spec. For example, one thing you want to test is that a client can start from an expired root, and go through expired intermediate roots before landing at the latest, (hopefully) unexpired one.

File: tests/e2e_test.go (hunk 1 above)
Comment (asraa) [reply tests/e2e_test.go:547]: Adding verification -- I'll follow-up with a test using 2+ thresholds, but FWIW with threshold 1 and two keys, it doesn't matter which key I rotate out for. (I think your 2 scenarios are equivalent modulo ordering?)

**Comment by asraa** [comment]
> For example, one thing you want to test is that a client can start from an expired root, and go through expired intermediate roots before landing at the latest, (hopefully) unexpired one.

Cosign does expired metadata update tests! https://github.com/sigstore/sigstore/blob/94e61d015fa729e7bdbb39c6d922122927ed5e20/pkg/tuf/client_test.go#L400

I'll add more tests here as well, specifically around "expired" root rotation.

**Review by Hayden-IO — APPROVED** [review]

**Review by dlorenc — APPROVED** [review]

**Pull request merged**